File: domain/src/main/java/com/study/platform/domain/post/model/PostSearchPort.java
package com.study.platform.domain.post.model;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.UUID;

public interface PostSearchPort {

    Page<PostSearchResult> search(String keyword, String techStack, StudyPostStatus status,
                                  int maxMembers, Pageable pageable);

    void index(StudyPost post);

    void indexAll(List<StudyPost> posts);

    void delete(UUID postId);
}

File: domain/src/main/kotlin/com/study/platform/domain/post/model/PostSearchPort.kt
package com.study.platform.domain.post.model

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import java.util.UUID

interface PostSearchPort {

    fun search(keyword: String?, techStack: String?, status: StudyPostStatus?, maxMembers: Int, pageable: Pageable): Page<PostSearchResult>
    fun index(post: StudyPost)
    fun indexAll(posts: List<StudyPost>)
    fun delete(postId: UUID)
}
